src/components/RegistrationDetailsPage.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import { withRouter } from 'react-router-dom'
+import { connect } from 'react-redux'
+import ReactDragList from 'react-drag-list'
+
+import registrationActions from '../reducers/actions/registrationActions'
+
+import Typography from '@material-ui/core/Typography'
+import { Input, Card, CardContent, Select, MenuItem } from '@material-ui/core'
+import TopicDialog from './TopicDialog'
+
+import './RegistrationDetailsPage.css'
+
+class RegistrationDetailsPage extends React.Component {
+  render() {
+    const registration = this.props.ownRegistration
+    const { student, preferred_topics, questions, createdAt } = registration
+    const { first_names, last_name, student_number, email } = student
+
+    let firstname = ''
+    if (first_names.includes('*')) {
+      firstname = first_names.split('*')[1]
+    }
+    firstname = firstname.split(' ')[0]
+
+    const submittedDate = createdAt.replace(/T/, ' ').substr(0, 16)
+
+    return (
+      <div>
+        <Typography variant="h4" gutterBottom>
+          Registration details
+        </Typography>
+        <Typography variant="body1" gutterBottom>
+          Registration date: {submittedDate}
+        </Typography>
+        <div>
+          <h2>User details</h2>
+          <Typography variant="body1" gutterBottom>
+            Name: {firstname} {last_name}
+            <br />
+            Student number: {student_number}
+            <br />
+            Email: {email}
+          </Typography>
+        </div>
+        <div>
+          <h2>Preferred Topics</h2>
+          <div className="dragndrop-container">
+            <div className="dragndrop-indexes-container">
+              {preferred_topics.map((topic, index) => {
+                return (
+                  <Card key={index} className="dragndrop-index">
+                    {index + 1}
+                  </Card>
+                )
+              })}
+            </div>
+            <ReactDragList
+              className="dragndrop-list"
+              handles={false}
+              dataSource={preferred_topics}
+              rowKey="id"
+              row={(topic) => {
+                return <TopicDialog topic={topic} key={topic.content.title} />
+              }}
+              disabled
+            />
+          </div>
+        </div>
+        <div>
+          <h2>Details</h2>
+          {questions.map((question, index) => {
+            return (
```

Comment:
Could this be extracted to its own component for clarity & more helpful debugging info when using React dev tools?

Reply:
Do you mean like to a const within the same file or as a separate <MappedQuestions /> component from completely differend file?

Reply:
Yeah at this point I'd just put them in the same file outside `RegistrationDetailsPage`. I see a few logical units in the rendered structure, maybe `<UserDetails />`, `<PreferredTopics />` and `<RegistrationAnswers />`?